dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoResource.java

```diff
@@ -644,6 +617,32 @@ public SourceFile dockerfile(@ApiParam(value = "Tool id", required = true) @Path
         return entryVersionHelper.getSourceFile(containerId, tag, FileType.DOCKERFILE);
     }
 
+    @GET
+    @Timed
+    @UnitOfWork
+    @Path("/{containerId}/verifiedSources")
+    @ApiOperation(value = "Get the corresponding Dockstore.cwl file on Github.", tags = { "containers" }, notes = "Does not need authentication", response = String.class)
+    public String verifiedSources(@ApiParam(value = "Tool id", required = true) @PathParam("containerId") Long containerId) {
+        Tool tool = toolDAO.findById(containerId);
+        Helper.checkEntry(tool);
+
+        Set<String> verifiedSourcesArray = new HashSet<>();
+
+        for (Tag tag : tool.getTags()) {
+            if (tag.isVerified()) {
+                verifiedSourcesArray.add(tag.getVerifiedSource());
+            }
+        }
+
+        JSONArray jsonArray = new JSONArray();
+        try {
+            jsonArray = new JSONArray(verifiedSourcesArray.toArray());
+        } catch (JSONException ex) {
+            ex.printStackTrace();
```

Comment:
Don't throw away the exception, this should probably be a CustomWebApplicationException, internal server error